localstack/services/apigateway/integration.py

```diff
@@ -147,7 +151,7 @@ def get_internal_mocked_headers(
             ]["AccessKeyId"]
         )
     else:
-        access_key_id = None
+        access_key_id = account_id
```

Comment:
Not sure about this one. It will probably break the IAM enforcement if there is no role specified, as in -ext, we test for the access key id being the `INTERNAL_AWS_ACCESS_KEY_ID` for IAM to see that the source principal is a service.

Reply:
@dfangl do we have the null `role_arn` just in case of -ext tests?

Reply:
There a basically two options to allow an apigateway integration for a service to work.
1) Use a role, which apigateway assumes, and performs the requests assumed as this role -> the permissions for the integrations are attached to the role
2) Use resource based policies, and attach the permissions to the target resource, and allow apigateway as service principal to access the resource.

This is basically  this if/else condition. In the second case, we currently assume that the INTERNAL_AWS_ACCESS_KEY_ID is set.

Reply:
@dfangl do we have something in specific that we can check to validate this fix for IAM enforcement in case no role is specified? I had already run the [-ext pipeline](https://github.com/localstack/localstack-ext/actions/runs/6876194122) and its green.

Reply:
@dfangl I have made the required changes, please review whenever possible.